spec/lower_level_api/integration/basic_publish_spec.rb

```diff
@@ -0,0 +1,23 @@
+require "spec_helper"
+
+describe Bunny::Channel, "#basic_publish" do
+  before(:all) do
+    @connection = Bunny.new(:user => "bunny_gem", :password => "bunny_password", :vhost => "bunny_testbed")
+    @connection.start
+  end
+
+  after :all do
+    @connection.close if @connection.open?
+  end
+
+  context "when the the connection isn't open" do
+    let(:queue_name) { "bunny.basic.publish.queue#{rand}" }
+
+    it "raises a ConnectionClosedError" do
+      ch = @connection.create_channel
+      # Assume the IO thread closed the connection due to exception
+      @connection.instance_variable_set("@status", :closed)
```

Comment:
Is the test case fine like this? or would you prefer a higher level toxiproxy test?

Reply:
You can simply close the connection to the same effect. A Toxiproxy test would be more useful.

Reply:
If you close the connection with session#close you would get the other exception ```Bunny::ChannelAlreadyClosed:
       cannot use a closed channel! Channel id: 1``` 
because channel knows it's closed. I'll look into a toxiproxy test case. Remove this one?